### PR TITLE
Region Growing (Shape Detection)

### PR DESCRIPTION
**Pull request by sgiraudot**
## Summary of Changes

New algorithm, alternative to RANSAC for plane detection. Pre-approved small feature.

## Release Management

* Affected package(s): `Point_set_shape_detection_3` (and `Point_set_processing_3` for point set structuring)
* [Small Feature](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Region_growing), pre-approved --Andreas Fabri 09:21, 7 July 2017 (CEST)

### CODE REVIEW

**Comment by lrineau** [comment]
There is a CMake error in Travis:
https://travis-ci.org/CGAL/cgal/jobs/251149426#L1506

File: Point_set_shape_detection_3/examples/Point_set_shape_detection_3/CMakeLists.txt [hunk 1]
```diff
@@ -23,12 +23,22 @@ if ( CGAL_FOUND )
...
   include_directories (BEFORE "../../include")
 
-  create_single_source_cgal_program( "efficient_RANSAC_basic.cpp" )
+  create_single_source_cgal_program( "shape_detection_basic.cpp" )
   create_single_source_cgal_program( "efficient_RANSAC_custom_shape.cpp" )
   create_single_source_cgal_program( "efficient_RANSAC_parameters.cpp" )
   create_single_source_cgal_program( "efficient_RANSAC_point_access.cpp" )
   create_single_source_cgal_program( "plane_regularization.cpp" )
 
+  find_package( TBB REQUIRED )
```
Comment (lrineau) [inline Point_set_shape_detection_3/examples/Point_set_shape_detection_3/CMakeLists.txt]: The Travis error is because TBB is not an optional dependency.

**Comment by lrineau** [comment]
This PR has conflicts, and last Travis build had errors.

**Comment by lrineau** [comment]
Ping!   (...conflicts, plus Travis errors)

**Comment by sgiraudot** [comment]
Rebased (FYI I fixed one final conflict from _Github_ and it created the latest commit).

**Comment by sloriot** [comment]
`Point_set_shape_detection_3/test/Point_set_shape_detection_3/test_scene.cpp` does not compile on my machine.

**Comment by sgiraudot** [comment]
I fixed the compilation of `test_scene` and `test_regularization`, but this later fails at execution, I'm investigating.

**Comment by sgiraudot** [comment]
Okay, the execution error is fixed too.

**Comment by sloriot** [comment]
Testsuite is still not good.

**Comment by sloriot** [comment]
See also the documentation [errors](https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.12-Ic-7/Point_set_shape_detection_3.log).

**Comment by sgiraudot** [comment]
There's an error in the plugin. I'll fix it.

**Comment by sloriot** [comment]
See also new warnings in Advancing front and PSP

**Comment by sgiraudot** [comment]
Some warnings remaining. I'm taking care of them.

**Pull request merged**